### ISSUE TITLE
Add a zoom in/out feature to Steam Client (Valve)

### ISSUE THREAD

**Issue by rnz** [ValveSoftware/steam-for-linux#6007]
#### Your system information

* Steam client version (build number or date): Any
* Distribution (e.g. Ubuntu): Any
* Opted into Steam client beta?: Yes
* Have you checked for system updates?: Yes

Text in the store (and other places) is too small. It seems to be browser based. Can't you just enable the zoom in zoom out feature (Ctrl +, Ctrl -). 
But Valve is ignoring users very long time - [Steam community](https://steamcommunity.com/discussions/forum/10/666826069015105830)

Skin subsystem - useless.

An example of how it looks on display with hi res:
![image](https://user-images.githubusercontent.com/851927/50576401-065dd380-0e22-11e9-9391-4d6688156f51.png)

**Comment by kisak-valve** [comment]
Related: https://github.com/ValveSoftware/steam-for-linux/issues/5427#issuecomment-378675490 and #5362

**Comment by TimTomTwo** [comment]
+1

**Comment by rnz** [comment]
Me when see into Steam:
![spunchbobcrying](https://user-images.githubusercontent.com/851927/56697891-ba001180-66f8-11e9-922a-0e696f72925c.gif)

**Comment by rnz** [comment]
And year later:
![will](https://user-images.githubusercontent.com/851927/56698374-61ca0f00-66fa-11e9-93c9-f066e3b90c43.gif)

**Comment by rnz** [comment]
and year again:
![crying1](https://user-images.githubusercontent.com/851927/56698388-73abb200-66fa-11e9-84e9-6f0f9c464986.gif)

**Comment by rnz** [comment]
and again:
![bolduin](https://user-images.githubusercontent.com/851927/56698414-8920dc00-66fa-11e9-9810-97482840791e.gif)

**Comment by rnz** [comment]
and again:
![spiderman](https://user-images.githubusercontent.com/851927/56698425-9342da80-66fa-11e9-845e-7f27020e7ee7.gif)

**Comment by rnz** [comment]
and again:
![crying2](https://user-images.githubusercontent.com/851927/56698443-9d64d900-66fa-11e9-8937-7f83b3406dd6.gif)

**Comment by rnz** [comment]
and again:
![interstellar](https://user-images.githubusercontent.com/851927/56698464-ace42200-66fa-11e9-936b-b2f63353844a.gif)

**Comment by rnz** [comment]
and again:
![gameoftrones](https://user-images.githubusercontent.com/851927/56698481-b5d4f380-66fa-11e9-9186-efdde9ddba78.gif)

**Comment by rnz** [comment]
Guys, you really think - people can read this?
[image](https://user-images.githubusercontent.com/851927/61181334-b3b16f80-a62d-11e9-81e6-2363f0076c17.png)

**Comment by matheusfillipe** [comment]
Same issue on ubuntu 18.04.3 LTS. Even though it's a 1080p monitor It's annoyingly too small.

**Comment by lammetje77** [comment]
while in the steam client go to the top left corner and open the drop down menu called steam

from there select settings and a new window will pop up

in there select the library tab

now you have 4 options : auto, small, medium & large

log out of steam relog back in and see which setting you like the most.

**Comment by TimTomTwo** [comment]
@lammetje77 
That's nice and all, but it doesn't solve the problem. It maybe changes something in the Library with all the games and stuff, but that's about it.
This option does nothing everywhere else (store page, browser, workshop, etc.). And a fixed zoom option, where you also have to restart/re-login is just bad design.

**Comment by matheusfillipe** [comment]
> while in the steam client go to the top left corner and open the drop down menu called steam
> 
> from there select settings and a new window will pop up
> 
> in there select the library tab
> 
> now you have 4 options : auto, small, medium & large
> 
> log out of steam relog back in and see which setting you like the most.

That surely makes it better but is still not zooming. I think what we want is just being able to hold control and scroll in or out to actually zoom, increasing or decreasing the whole width of the steam Store and library, and that keeps saved for the next steam run.

**Comment by lammetje77** [comment]
> That's nice and all, but it doesn't solve the problem. It maybe changes something in the Library with all the games and stuff, but that's about it.
> This option does nothing everywhere else (store page, browser, workshop, etc.). And a fixed zoom option, where you also have to restart/re-login is just bad design.

not sure what you did wrong because if i do this i can read all the pages you are listing that don't change for you.

but i agree with you that there is a huge flaw in there design that we can not normally zoom in/out like we want to.


but then again that is not my only issue with the steam app
very often when i am scrolling up or down with the mouse wheel it just stops
only fix i know for it atm is to just right click any where on the page to get the pop up window and then i can scroll around again

**Comment by TimTomTwo** [comment]
@lammetje77 
I'm 100% sure that these options in the "Library"-tab only changes the size of the icons and text in the "Library" and nothing else. 
I double checked this right now and I didn't need to re-login to see the change in the Library page. (but I re-logged in anyways)
All the other pages I listed where exactly the same size.

**Comment by rnz** [comment]
Year later: 
![un23nn0](https://user-images.githubusercontent.com/851927/74745362-d61b0380-5274-11ea-9c2c-55d9db376c4a.gif)

**Comment by rnz** [comment]
2120 year - Steam still doesn't have zoom feature.

**Comment by matheusfillipe** [comment]
> Year later:
> ![un23nn0](https://user-images.githubusercontent.com/851927/74745362-d61b0380-5274-11ea-9c2c-55d9db376c4a.gif)

LOL! But that's so true. Steam basically acts like if it was a browser but can't have a feature like ctrl+scroll or ctrl+ + ...

**Comment by Typ33** [comment]
^ 2120 steam update.
Big Text Mode added.

**Comment by rnz** [comment]
Starship Enterprise makes its first space flight at stardate  2148 then Valve stiil can't add a zoom to the Steam client.

**Comment by IoIxD** [comment]
I got it! It's not the best solution but it certainly works!

After downloading a High-DPI friendly skin like Metro, modify the `resource/webkit.css` file in the downloaded skin folder and add 
```
:root {
	zoom:  1.5;
}
```
to the top. This will force the webpages to zoom in. Change the zoom parameter to your liking, of course.
Alternatively you could just make a blank skin and add this, but Metro fixes the ui itself anyways so I would recommend just getting that.

**Comment by rnz** [comment]
USS Enterprise (NCC-1701) is already under construction on the ropes... But Valve still can't add a zoom to the Steam client.

**Comment by komprimiert** [comment]
@lolxD how do you do this on windows? because there is no webkit.css in the resource folder for me

**Comment by mmatis** [comment]
Sadly, the webkit.css trick no longer works on the current Steam Beta. I can't believe the UI accessibility is still a problem in Steam after all these YEARS. Valve, please stop assuming what other people want/need. It's called Human Centered Design. By giving us a font size / UI Scale slider, you enable your customers adapt your UI to their individual needs. This is an accessibility issue for people with low vision, AND a usability issue for people on high-res displays!

**Comment by Xenophobe-01** [comment]
This is a standard feature in most browsers these days. Since Valve is apparently taking steam full CEF (chromium embedded framework) and zoom level is an option with CEF, there is no obvious reason this feature should not be implemented in the desktop small/large modes.

**Comment by Ex3cuteR00t** [comment]
Same here, the webkit.css no longer fully works. Every page on steam (store, community, profile etc) uses the zoom apart from library, which is the only page on steam i'd want the zoom to be used in.

**Comment by sergiohzph** [comment]
Are you fucking telling me they've been asking for this **SINCE 2013** and they still haven't done this? What the fuck are you on about Valve? You suck. You update whatever the fuck you want whenever the fuck you want, but you don't fucking listen to those of us who give you money.

No one asked for a Steam Deck, no one asked for the points system in the shop, no one asked for SteamVR, no one asked for Steam Labs, no one asked for any of the bullshit you've released. Instead what people have asked for, you don't fucking want to give us. You're fucking disgusting, you only care about money, you don't give a fuck about your users.

**Comment by loulloko** [comment]
Steam just had a major UI update, there still isn't an option to zoom in. They did remove skins though lol

**Comment by Xenophobe-01** [comment]
You can now set a custom zoom by adding the command line switch `-forcedesktopscale X` where X can be 2 for 200% or 1.25 for 125% or even something like 1.42 if 142% works for you. Not sure what the cap is on it, but 3 was too big on a 55"4K tv.

**Comment by GownerCode** [comment]
> You can now set a custom zoom by adding the command line switch `-forcedesktopscale X`

This does not work for me in the newest Steam beta version.

**Comment by cosclouds2** [comment]
they never listen

**Comment by hianick** [comment]
I would also love the opposite since it can only get so small on a 1080p monitor, modern UIs are soooo big these days. Accessibility would be great in both directions.

**Comment by Xenophobe-01** [comment]
> I would also love the opposite since it can only get so small on a 1080p monitor, modern UIs are soooo big these days. Accessibility would be great in both directions.

Try adding the command line switch -forcedesktopscale X, where X is 0.8 or 0.75 or 0.5, etc.  I've not needed to go below 1.0 myself, but it may work.

**Comment by RokeJulianLockhart** [comment]
> https://github.com/ValveSoftware/steam-for-linux/issues/6007#issue-395112122

@rnz, I don't believe that a scale feature should be added to remediate this problem - I believe that inheriting the system text size would be better. However, if buttons are also inconsistently small, then perhaps this is more of a universal solution.
